app/scripts/lib/WeakDomainProxyMap.ts

```diff
@@ -0,0 +1,58 @@
+type WeakReffedValueObj = {
```

Comment:
Do you think it's worth making this file more generic?

```
type WeakRefObject<T> = {
  [K in keyof T]: WeakRef<T[K]>;                            
};

...

export class WeakRefObjectMap<T> {
  map: Map<string, WeakRefObject<T>>;
  
...
```

Reply:
Yeah we sure could! I'll go ahead and do that!